src/main/kotlin/com/criptext/mail/scenes/signin/holders/ConnectionHolder.kt

```diff
@@ -75,25 +75,24 @@ class ConnectionHolder(val view: View, val username: String,
 
     fun setProgress(message: UIMessage, progress: Int) {
         textViewStatus.text = view.context.getLocalizedUIMessage(message)
-        timer.start(50, Runnable {
-            val stepProgress = progressBar.progress + 1
-            if(stepProgress <= progress) {
-                updateProgress(stepProgress)
-            }
-        })
+
+        val anim = ValueAnimator.ofInt(progressBar.measuredWidth, (progress * 9))
+        anim.addUpdateListener { valueAnimator ->
+            val `val` = valueAnimator.animatedValue as Int
+            val layoutParams = progressBar.layoutParams
+            layoutParams.width = `val`
+            progressBarNumber.text = (`val`/9).toString().plus("%")
+            progressBar.layoutParams = layoutParams
+        }
+        anim.duration = 1000
```

Comment:
Create a generic function that takes a progress bar and returns an `ObjectAnimator`